sklearn/compose/_column_transformer.py

```diff
@@ -234,6 +236,8 @@ def _iter(self, X=None, fitted=False, replace_strings=False):
                         check_inverse=False)
                 elif trans == 'drop':
                     continue
+                elif hasattr(column, '__len__') and len(column) == 0:
```

Comment:
```py
>>> len(np.zeros((10, 0)))
10
```

Isn't that a problem in this context? Use `.size`?

Reply:
`columns` should not be multi-dimensional, and `.size` is not available for lists. And we can also pass a single scalar (int or string), so the `hasattr(column, '__len__')` was actually a test for `column` being a Sequence. Maybe that would be better done with using `isinstance(.., collections.Sequence)`?

But, now I think about it, we should also catch the case of a boolean array of all False I think, for which the check of zero length will also not work.